From 795a55d94ccbe3306c683f18e3227196d9bcc4dd Mon Sep 17 00:00:00 2001
From: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
Date: Wed, 29 May 2013 16:53:22 +0200
Subject: [PATCH] MPP: HOST_RUNNING, HDD_USABLE

Set MPP36 as gpio direction output HOST_RUNNING.
Set MPP37 as gpio direction input HDD_USABLE.

Signed-off-by: Mario Schuknecht <mario.schuknecht@dresearch-fe.de>
---
 arch/arm/mach-kirkwood/openrd-setup.c |   39 +++++++++++++++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/arch/arm/mach-kirkwood/openrd-setup.c b/arch/arm/mach-kirkwood/openrd-setup.c
index c1cdce5..1adb77b 100644
--- a/arch/arm/mach-kirkwood/openrd-setup.c
+++ b/arch/arm/mach-kirkwood/openrd-setup.c
@@ -135,6 +135,10 @@ static unsigned int openrd_mpp_config[] __initdata = {
 	MPP28_GPIO,
 	MPP29_GPIO,
 	MPP34_GPIO,
+	MPP35_GPIO,
+	MPP36_GPIO,
+	MPP37_GPIO,
+	MPP38_GPIO,
 	0
 };
 
@@ -218,6 +222,39 @@ static int __init uart1_mpp_config(void)
 	return 0;
 }
 
+static int __init hdd_host_mpp_config(void)
+{
+	if (gpio_request(36, "HOST_RUNNING")) {
+		printk(KERN_ERR "GPIO request failed for HOST_RUNNING"
+				", gpio: 36\n");
+		return -EIO;
+	}
+	if (gpio_direction_output(36, 1)) {
+		printk(KERN_ERR "GPIO direction output failed for HOST_RUNNING"
+				", gpio: 36\n");
+	} else if (gpio_export(36, 0)) {
+		printk(KERN_ERR "GPIO export failed for HOST_RUNNING"
+				", gpio# 36\n");
+	}
+
+	if (gpio_request(37, "HDD_USABLE")) {
+		printk(KERN_ERR "GPIO request failed for HDD_USABLE"
+				", gpio# 37\n");
+		gpio_free(36);
+		return -EIO;
+	}
+
+	if (gpio_direction_input(37)) {
+		printk(KERN_ERR "GPIO direction input failed for HDD_USABLE"
+				", gpio# 37\n");
+	} else if (gpio_export(37, 0)) {
+		printk(KERN_ERR "GPIO export failed for HDD_USABLE"
+				", gpio# 37\n");
+	}
+
+	return 0;
+}
+
 static void __init openrd_init(void)
 {
 	/*
@@ -252,6 +289,8 @@ static void __init openrd_init(void)
 		kirkwood_audio_init();
 	}
 
+	hdd_host_mpp_config();
+
 	if (uart1 <= 0) {
 		if (uart1 < 0)
 			printk(KERN_ERR "Invalid kernel parameter to select "
-- 
1.7.9.5

